showcase-docs/dynamic-plugins.md

```diff
@@ -0,0 +1,271 @@
+
+```ts
+export * from './dynamic/index';
+```
+
+### Exporting the plugin as a dynamic plugin package
+
+Once the code changes are done, the plugin can be exported as a dynamic plugin package, using the `export-dynamic` script entry:
+
+```bash
+yarn export-dynamic
+```
+
+The resulting package will be located in the `dist-dynamic` sub-folder of the plugin folder. It is renamed by adding the `-dynamic` suffix to the plugin name.
+
+This allows packing it with `npm pack`, or publishing it to an npm registry.
+
+The dynamic export mechanism identifies private, non-backstage dependencies, and sets the `bundleDependencies` field in the `package.json` file for them, so that the dynamic plugin plackage can be published as a self-contained package, along with its private dependencies bundled in a private `node_modules` folder.
+
+Common backstage dependencies, expected to be in the backstage backend application, are not bundled in the dynamic plugin but rather changed as peer dependencies, so that they can be shared with the backstage backend application.
+
+#### About embedding dependencies in the plugin package
+
+The `export-dynamic-plugin` command can also embed some dependencies in the dynamic plugin package, by using the `--embed-package` option. For example:
+
+```bash
+yarn export-dynamic --embed-package @roadiehq/scaffolder-backend-module-http-request
+```
+
+This merges the code of the specified dependencies in the generated dynamic plugin code, and updates the `package.json` file to hoist their dependencies to the top level.
+
+This is useful when wrapping a third-party plugin to make it compatible with the dynamic plugin support, as explained in the next section.
+
+#### Wrapping a third-party plugin to add dynamic plugin support
+
+In order to add dynamic plugin support to a third-party plugin, without touching the third-party plugin source code, a wrapper plugin can be created that will:
+
+- import the third-party plugin,
+- include the additions described above,
+- export it as a dynamic plugin.
+
+An example of such a wrapper plugin can be found in the [dynamic backend plugin showcase repository](https://github.com/janus-idp/dynamic-backend-plugins-showcase/tree/main/plugins/scaffolder-backend-module-http-request-wrapped): it wraps the `@roadiehq/scaffolder-backend-module-http-request` package to make it compatible with the dynamic plugin support and, through the use of the `--embed-package` option in the [`export-dynamic` script](https://github.com/janus-idp/dynamic-backend-plugins-showcase/blob/d254c065764ab49289c2bcaad9fd996d49003f9d/plugins/scaffolder-backend-module-http-request-wrapped/package.json#L21), embeds the wrapped plugin code in the generated code and hoist its `@backstage` dependencies as peer dependencies in the resulting dynamic plugin.
+
+## Installing a dynamic plugin package in the showcase
+
+### Local configuration
+
+- Create a `dynamic-plugins-root` folder at the root of the showcase application repository.
```

Comment:
Should we point them to `./install-dynamic-plugins.py` instead?

Reply:
I don't think so. There's not use for this. This seems over-complicated for the local case.